airflow/providers/snowflake/hooks/snowflake.py

```diff
@@ -310,14 +309,14 @@ def run(
 
                     self.log.info("Rows affected: %s", cur.rowcount)
                     self.log.info("Snowflake query id: %s", cur.sfqid)
-                    self.query_ids.append(cur.sfqid)
+                    query_ids.append(cur.sfqid)
 
             # If autocommit was set to False for db that supports autocommit,
             # or if db does not supports autocommit, we do a manual commit.
             if not self.get_autocommit(conn):
                 conn.commit()
 
-        return execution_info
+        return execution_info, query_ids
```

Comment:
According to the method description of the `DBAPiHook` base class, `DBAPiHook.run` should return the results of the operation. It shouldn't return a tuple, one of the elements of which is the result of the operation.
https://github.com/apache/airflow/blob/b00ce4592fb21bd4cf4ff6227fa5a498eb357faf/airflow/hooks/dbapi.py#L192
Besides, it is a breaking change that we should avoid.
If we really want to introduce this feature, then we should do it as a new method.

Reply:
yup it seems like we cannot practically make this change.  it seems that the issue i created (https://github.com/apache/airflow/issues/20398) is not a good issue, since there's not really a practical way to get around it.